code/modules/projectiles/projectile/beams.dm

```diff
@@ -38,6 +38,17 @@
 	weaken = 5
 	stutter = 5
 
+
+/obj/item/projectile/beam/immolator
+	name = "immolation beam"
+
+/obj/item/projectile/beam/immolator/on_hit(var/atom/target, var/blocked = 0)
+	. = ..()

```

Comment:
the base proc returns 0/1, some (but not all) overrides also return 0/1, seems logical to assume something is or WAS expecting a return value from this proc.